tests/model_selection/test_incremental.py

```diff
@@ -731,3 +822,17 @@ def test_search_invalid_patience(c, s, a, b):
     search = IncrementalSearchCV(model, params, patience=False, max_iter=10)
     yield search.fit(X, y, classes=[0, 1])
     assert search.history_
+
+
+def test_warns_scores_per_fit():
+    X, y = make_classification(n_samples=100, n_features=5, chunks=10)
+
+    params = {"value": np.random.RandomState(42).rand(1000)}
+    model = ConstantFunction()
+
+    with pytest.warns(UserWarning, match="deprecated since Dask-ML v1.4.0"):
+        IncrementalSearchCV(model, params, scores_per_fit=2)
+    if datetime.now() >= datetime(2021, 1, 1):
```

Comment:
I'm unsure about including this date in the test. How should this be handled? I don't want to let this be deprecated forever.

Reply:
This can be removed. We'll update the test when the deprecation is enforced.

To make sure I understand, `scores_per_fit` and `fits_per_score` have identical semantics, we're just giving it a better name? We might just make an API breaking change in that case.

Reply:
Yup, the renaming is purely cosmetic. Should I leave the warning in or raise a `ValueError`?

Reply:
IMO we either just make the change (and remove this test) or we leave in the deprecation and assert that a warning is raised (and remove this date-dependent assert).

Reply:
I'd rather leave a warning in and remove it shortly. I've pushed that change, and it looks like the CI failure is unrelated (documentation warnings in `hyper-parameter-search.rst`, which is unmodified in this PR).